src/dfx/src/lib/operations/canister/deploy_canisters.rs

```diff
@@ -113,14 +116,19 @@ pub async fn deploy_canisters(
         {
             call_sender
         } else {
-            let wallet = get_or_create_wallet_canister(
+            match get_or_create_wallet_canister(
                 env,
                 env.get_network_descriptor(),
                 env.get_selected_identity().expect("No selected identity"),
             )
-            .await?;
-            proxy_sender = CallSender::Wallet(*wallet.canister_id_());
-            &proxy_sender
+            .await
+            {
+                Ok(wallet) => {
+                    proxy_sender = CallSender::Wallet(*wallet.canister_id_());
+                    &proxy_sender
+                }
+                Err(_) => call_sender,
```

Comment:
Is it appropriate to use `call_sender` in all error cases, or just some of them?  For example, if `create_wallet` created the wallet, installed the code, but failed in `wallet_store_wallet_wasm`, should this swallow the error and use call_sender?  Or is there some smaller set of error conditions under which this should happen.